File: src/List/SingleLinkedList/AddTwoNumbers.java
package List.SingleLinkedList;

public class AddTwoNumbers {
}


// Definition for singly-linked list.
class ListNode {
     int val;
     ListNode next;
     ListNode() {}
     ListNode(int val) { this.val = val; }
     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 }

class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode cur1=l1;
        ListNode cur2=l2;
        int cout=0;
        ListNode cur3=new ListNode();
        ListNode result=cur3;
        while(cur1!=null||cur2!=null)
        {
            int val1,val2;
            val1=(cur1==null)?0:cur1.val;
            val2=(cur2==null)?0:cur2.val;
            cur3.val=(val1+val2+cout)%10;
            if(val1+val2+cout>=10)
                cout=1;
            else
                cout=0;
            cur1=(cur1==null)?null:cur1.next;
            cur2=(cur2==null)?null:cur2.next;
            if(cur1!=null||cur2!=null){
                cur3.next=new ListNode();
                cur3=cur3.next;
            }
        }
        if(cout==1)
            cur3.next=new ListNode(1);
        return result;
    }
}

File: list/AddTwoNumbers.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        ans = ListNode()
        pointer = ans
        Ci = 0
        while l1 is not None or l2 is not None:
            twoSum = (l1.val if l1 else 0) + (l2.val if l2 else 0) + Ci
            pointer.next = ListNode(val=twoSum % 10)
            Ci = 1 if twoSum >= 10 else 0
            pointer = pointer.next
            l1 = l1.next if l1 else None
            l2 = l2.next if l2 else None
        if Ci == 1:
            pointer.next = ListNode(val=1)
        return ans.next
